src/pybind/mgr/telemetry/module.py

```diff
@@ -768,6 +766,14 @@ def handle_command(self, inbuf, command):
             return (-errno.EINVAL, '',
                     "Command not found '{0}'".format(command['prefix']))
 
+    def on(self):
+        self.set_module_option('enabled', True)
+        self.set_module_option('last_opt_revision', REVISION)
+
+    def off(self):
+        self.set_module_option('enabled', False)
+        self.set_module_option('last_opt_revision', 1)
```

Comment:
What about doing the same kind of refactoring you did here to the report gathering methods, so the whole report is generated in the telemetry module, and the dashboard just only needs to expose it?

Reply:
I'm not really sure what you mean. Are you talking about adding a method that returns the result of `compile_report` and `gather_device_report` (if exists) in one object?